src/Build.UnitTests/Scanner_Tests.cs

```diff
@@ -100,6 +100,21 @@ public void IllFormedProperty()
             Assert.Equal("IllFormedPropertyOpenParenthesisInCondition", lexer.GetErrorResource());
         }
 
+        /// <summary>
+        /// Tests the whitespace errors case
+        /// </summary>
+        [Fact]
+        public void WhitespaceProperty()
+        {
+            Scanner lexer = new Scanner("$(x )", ParserOptions.AllowProperties);
+            AdvanceToScannerError(lexer);
+            Assert.Equal("IllFormedPropertyWhitespaceInCondition", lexer.GetErrorResource());
+
+            lexer = new Scanner("$( x)", ParserOptions.AllowProperties);
```

Comment:
is it worth adding a test for "$(x yz" and "$(x y z")?

Reply:
do you really mean `"$(x yz" and "$(x y z")` or actually `"$(x yz)" and "$(x y z)"`?

Reply:
The latter 😊